cli/src/commands/create/trigger_exception.rs

```diff
@@ -0,0 +1,46 @@
+use crate::printer::Printer;
+use anyhow::{Context, Result};
+use log::info;
+use reinfer_client::{
+    Client, CommentUid, TriggerException, TriggerExceptionMetadata, TriggerFullName,
+};
+use structopt::StructOpt;
+
+#[derive(Debug, StructOpt)]
+pub struct CreateTriggerExceptionArgs {
+    #[structopt(short = "t", long = "trigger")]
+    /// The dataset name or id
+    trigger: TriggerFullName,
+
+    #[structopt(long = "type")]
+    /// The type of exception. Please choose a short, easy-to-understand string such as "No Prediction".
+    r#type: String,
```

Comment:
What's the reason for using a raw string here and not something like `&str`? I'm guessing it's because the values that we're passing in aren't going to be around long enough, so we have to take ownership?

Reply:
There's a technical answer about having to add lifetimes to any borrowed values in structs, which structopt doesn't support AFAIK, but the question really boils down to - who owns the data? When I input a `--type "Sad Violin"` flag, who owns `"Sad Violin"`?

You can see the argument parsing call here, right at the start of the program:
https://github.com/reinfer/cli/blob/e75c12914a05d0f371c9207434211e23ecc04e62/cli/src/main.rs#L148

The output is the variable `args`, and **there are no other variables defined before it**. So it would be hard to reference data that's owned by another variable! We could be referencing data in the binary, like a `static`, but that's not the case. We could be referencing data backed by an external resource, but again that's not the case. When we parse the CLI args, we consume them and spit out `args`. So, it **must own all the data**.

That's why we have to use an owned value, like `String`. (I'm not sure what you mean by 'raw string'? The `r#...` syntax is a [raw identifier](https://doc.rust-lang.org/reference/identifiers.html#raw-identifiers) that's used when the identifier would otherwise be a reserved Rust keyword).